src/dispatch/static/dispatch/src/api.js

```diff
@@ -20,4 +20,18 @@ instance.interceptors.request.use(
   }
 )
 
+instance.interceptors.response.use(
```

Comment:
Today, we check for auth on route load (instead of 401) and then have a timer telling a user to refresh before their token expires. 
See here:
https://github.com/Netflix/dispatch/blob/develop/src/dispatch/static/dispatch/src/router/index.js#L137
and
https://github.com/Netflix/dispatch/blob/develop/src/dispatch/static/dispatch/src/auth/store.js#L31

I'm not sure how that fits in with the login feature? Perhaps we set the same expiration but send users to /login when basic authentication is enabled?

We will also need a way to tell the frontend to use PKCE or basic authentication via the `DISPATCH_AUTHENTICATION_PROVIDER_SLUG` here:

https://github.com/Netflix/dispatch/blob/develop/src/dispatch/static/dispatch/src/router/index.js#L31

Reply:
I moved the auth check, in the router/index.js although I kept the check on the responses so I can effectively logout the user, this should not interact with any other auth provider, unless there are 401 returned (which is not a bad intent to logout user if 401 is received)

This should not interact with the current login feature. Exactly when no auth provider slug is provided basic auth will be used and this handler will be accessed "loginBasic" from router/index.js.

I decided to let it empty and modify value of `VUE_DISPATCH_AUTHENTICATION_PROVIDER_SLUG` to set a default value in case no slug is present